File: 프로그래머스/unrated/135808. 과일 장수/과일 장수.java
import java.util.*;

class Solution {
    public int solution(int k, int m, int[] score) {
        int answer = 0;
        int cnt = score.length / m;
        
        Arrays.sort(score);
        
        for(int i = score.length - 1; i > 0; i -= m){
            if(i - m + 1 < 0){
                break;
            }
            else{
                answer += (score[i - m + 1] * m);
            }
        }    

        return answer;
    }
    
    public void printArr(int[] arr){
        int i;
        int length = arr.length;
        
        for (i = 0; i < length; i += 1) {
            System.out.printf("%d ", arr[i]);
        }
        System.out.println();

    }
}

File: 프로그래머스/1/135808. 과일 장수/과일 장수.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(int k, int m, vector<int> score) {
    int answer = 0;
    
    int n = score.size();
    int boxCnt = n / m;
    int add = n - m * boxCnt;
    
    int Pos = add;
    
    sort(score.begin(), score.end());

    for(Pos; Pos < n; Pos = Pos + m){
        int p = score[Pos];
        answer += p * m;
    }
        
    
    return answer;
}